### ISSUE TITLE
Scaling of default priors

### ISSUE THREAD

**Issue by fweber144** [paul-buerkner/brms#1065]
According to [these lines](https://github.com/paul-buerkner/brms/blob/34e769072e4853c86c2fbf150f603b0e2f5feb2b/R/priors.R#L1090-L1093), it seems like that for a non-factor-like `y` with a link function from <https://github.com/paul-buerkner/brms/blob/34e769072e4853c86c2fbf150f603b0e2f5feb2b/R/priors.R#L1083>, the scale of the default Student-t prior is set to the maximum of `scale` and `scale_y`. Wouldn't it make more sense to set the scale of that Student-t prior to `scale * scale_y` as done in [rstanarm](https://mc-stan.org/rstanarm/articles/priors.html#default-priors-and-scale-adjustments)? Perhaps there was a good reason to choose the maximum, I just don't see it and wanted to highlight this in case it was not intended.

**Comment by paul-buerkner** [comment]
Why would you say it makes more sense?

**Comment by fweber144** [comment]
Consider for example a response `y` with `loc_y = median(y) = 0` and `scale_y = mad(y) = 3` which is modelled as `y ~ normal(Intercept, sigma)` (and this also being the true data-generating model for some true `Intercept` and some true `sigma`).

If *a priori* `Intercept ~ student_t(3, loc_y, max(2.5, scale_y)) = student_t(3, 0, 3)`, then an extreme `Intercept` (i.e. at the boundary of the central 90% prior interval) would be `3 * qt(0.95, df = 3) * c(-1, 1) = c(-7.06..., 7.06...)` which is slightly more than `2 * scale_y * c(-1, 1)`. In my opinion, this is too narrow for a weakly informative default prior. For example, this "extreme" prior value may easily lie within the empirical distribution of `y` (which is only a weak argument for the narrowness of the prior, but I thought it might give some intuition).

In contrast, if *a priori* `Intercept ~ student_t(3, loc_y, 2.5 * scale_y) = student_t(3, 0, 7.5)`, then an extreme `Intercept` would be `7.5 * qt(0.95, df = 3) * c(-1, 1) = c(-17.65..., 17.65...)` which is about `6 * scale_y * c(-1, 1)`. This should be ok for a weakly informative prior. For example, that value typically lies far outside of the empirical distribution of `y`.

The difference of the two approaches should be especially relevant for small samples where `scale_y` may be a bad estimate of the true standard deviation and `loc_y` a bad estimate of the true mean. Maybe I can also create a more illustrative example with simulated data.

I haven't spent too many thoughts yet on other parameters which might be affected by `def_scale_prior()` (e.g. group-level standard deviations). Originally, I was also thinking of regression coefficients `b` where this default scaling might be too informative, but I had forgotten that the default prior for `b` is flat, so this issue is irrelevant for `b`.

**Comment by fweber144** [comment]
Here is a more specific example with simulated data:
```r
library(ggplot2)
set.seed(831490948)
y <- rnorm(5, sd = 10)
loc_y <- median(y)
scale_y <- mad(y)
set.seed(2143203840)
y_prior_max <- brms::rstudent_t(1e4, df = 3, mu = loc_y, sigma = max(2.5, scale_y))
qplot(y_prior_max, geom = "density") +
  geom_vline(xintercept = 0, color = "forestgreen") +
  geom_vline(xintercept = y, color = "firebrick")
brms::pstudent_t(0, df = 3, mu = loc_y, sigma = max(2.5, scale_y))
# [1] 0.9345415
set.seed(2143203840)
y_prior_multiply <- brms::rstudent_t(1e4, df = 3, mu = loc_y, sigma = 2.5 * scale_y)
qplot(y_prior_multiply, geom = "density") +
  geom_vline(xintercept = 0, color = "forestgreen") +
  geom_vline(xintercept = y, color = "firebrick")
brms::pstudent_t(0, df = 3, mu = loc_y, sigma = 2.5 * scale_y)
# [1] 0.765279

```
The two ggplots may be found below. Above, I added `pstudent_t()` calls with their results to quantify the "extremeness" of the true intercept under the two priors.
![prior_max](https://user-images.githubusercontent.com/55132727/104095821-3ce44980-5299-11eb-8161-77a5ec3d4bf5.png)
![prior_multiply](https://user-images.githubusercontent.com/55132727/104095829-48d00b80-5299-11eb-954a-1e290a4c1957.png)

**Comment by paul-buerkner** [comment]
Thank you for your elaboration. I understand the point, but I am not (yet) convinced this is actually a problem. I would consider the current default prior to be problematic if one can create data sets and a corresponding model where the default priors are actually informative and thus change the posterior to a relevant degree. 

To be clear, I am not saying the current default prior is great, nor that it is better than what you are proposing. Rather, in order to make a change to the default priors now, instead of when I make a major overhaul over brms' prior system anyway, I would need to see examples where the current choice is actually problematic.

**Comment by fweber144** [comment]
Yeah, I understand. No problem. Perhaps with time going by I'll come across a better example (or improve my existing example). The rstanarm authors who made the decision to multiply `scale_y` by `scale` (2.5) might also have had more formal arguments.

**Comment by fweber144** [comment]
Do you want me to close this issue for now?

**Comment by paul-buerkner** [comment]
Ok, I would leave it open. It can be used as a starting point for dicussion on more principled default priors. Thank you for taking the time to think about these issues!

**Comment by fweber144** [comment]
For the example above with the two scale approaches shown there (taking the maximum or multiplying), this would be the marginal posterior for the intercept (with `y`, `loc_y`, and `scale_y` as defined above):
```r
library(brms)
options(mc.cores = 4)

bfit_prior_max <- brm(
  formula = y ~ 1,
  data = data.frame(y),
  seed = 65481213
)
mcmc_plot(bfit_prior_max, type = "dens", pars = "Intercept")
posterior_summary(bfit_prior_max, pars = "Intercept", robust = TRUE)
#              Estimate Est.Error      Q2.5      Q97.5
# b_Intercept -4.904879  2.102052 -9.214552 -0.5006225

bfit_prior_multiply <- brm(
  formula = y ~ 1,
  data = data.frame(y),
  prior = set_prior(
    paste0("student_t(3, ", round(loc_y, 1), ", ", 2.5 * round(scale_y, 1), ")"),
    class = "Intercept"
  ),
  seed = 65481213
)
mcmc_plot(bfit_prior_multiply, type = "dens", pars = "Intercept")
posterior_summary(bfit_prior_multiply, pars = "Intercept", robust = TRUE)
#             Estimate Est.Error      Q2.5    Q97.5
# b_Intercept -4.23017  2.573935 -9.948334 1.375459

```

When taking the maximum:
![post_dens_max](https://user-images.githubusercontent.com/55132727/104700847-139e4000-5715-11eb-8aed-83929539359c.png)

When multiplying:
![post_dens_multiply](https://user-images.githubusercontent.com/55132727/104700897-244eb600-5715-11eb-99d5-075cafd886e3.png)

So there's indeed some impact on the posterior (in particular, have a look at the central 95% posterior interval). But I'm not sure if this is what you meant by

> I would consider the current default prior to be problematic if one can create data sets and a corresponding model where the default priors are actually informative and thus change the posterior to a relevant degree.

**Comment by paul-buerkner** [comment]
Thank you! I think I do see that point now better thanks to your examples. I will think about it a little more but will probably implement your, or a similar approach.
